### PR TITLE
Issue #11865: Resolve link specefied as plain text idea violations 

### PR DESCRIPTION
**Pull request by Vyom-Yadav**
#11865 

`no_old_refs.sh` was also modified to support anchor tags.

### CODE REVIEW

**Comment by nick-mancuso** [comment]
Github, generate site

File: .ci/no_old_refs.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ if [ -n "$PR_HEAD_REPO_NAME" ]; then
 fi
 
 # collect issues where full link is used
-grep -IPonr "(after|[Tt]il[l]?) $GITHUB_HOST/[\w.-]+/[\w.-]+/issues/\d{1,5}" . \
+grep -IPonr "(after|[Tt]il[l]?) (<a href=\")?$GITHUB_HOST/[\w.-]+/[\w.-]+/issues/\d{1,5}" . \
```
Comment (romani) [inline .ci/no_old_refs.sh:23]: if  it optional , why we need this in regexp ?
lets keep regexp simple, please restore to original version.

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: .ci/no_old_refs.sh (hunk 1 above)
Comment (Vyom-Yadav) [reply .ci/no_old_refs.sh:23]: No violation:
`// Until https://github.com/checkstyle/checkstyle/pull/12048`

IDEA inspections violation (for javadoc):
```
/**                                                         
 * Until https://github.com/checkstyle/checkstyle/pull/12048
 */                  
```
Need to change it to:
```
/**                                                                              
 * Until <a href="https://github.com/checkstyle/checkstyle/pull/12048">#12048</a>
 */                                                                              
```

Comment (Vyom-Yadav) [reply .ci/no_old_refs.sh:23]: It is used in some cases, in some, it isn't, so we need to keep it.

Comment (romani) [reply .ci/no_old_refs.sh:23]: .ci/no_old_refs.sh does not care about IDEA violations, it just search all urls.

if you remove change in this file, what bad will happen ?

Comment (Vyom-Yadav) [reply .ci/no_old_refs.sh:23]: It cannot find `Until <a href="https://github.com/checkstyle/checkstyle/pull/12048">#12048</a>` without the modification.

> what bad will happen ?

If we have an old ref, it won't be able to determine it.

Comment (romani) [reply .ci/no_old_refs.sh:23]: I see

**Review by romani — APPROVED** [review]
ok to merge

**Review by nick-mancuso — APPROVED** [review]

**Pull request merged**